flink-core/src/main/java/org/apache/flink/configuration/DelegatingConfiguration.java

```diff
@@ -178,14 +178,19 @@ public String toString() {
 
 	@Override
 	public Set<String> keySet() {
+
 		final HashSet<String> set = new HashSet<String>();
-		final int prefixLen = this.prefix == null ? 0 : this.prefix.length();
 
 		for (String key : this.backingConfig.keySet()) {
-			if (key.startsWith(this.prefix)) {
+
+			if (this.prefix == null) {
```

Comment:
Can we move the condition out of the loop? The condition does not change over different iteration so we don't have to perform it every time.